twine/auth.py

```diff
@@ -0,0 +1,80 @@
+import warnings
+import getpass
+import functools
+
+import keyring
+import munch
+
+from . import utils
+from . import exceptions
+
+
+class Resolver:
+    def __init__(self, settings, username=None, password=None):
```

Comment:
Could you add type annotations to this and the methods? I know it's not enforced yet, and it could certainly be a separate PR, but it'd be nice to keep the existing type coverage.

Reply:
Yes, perhaps. You may have noticed that `settings` is currently duck-typed. It accepts any object with a `.system` and `.config` properties, including the `twine.settings.Settings` object. Adding typing for this will require creation of new classes to make this interface explicit.

Reply:
I believe I've added the necessary type annotations. Some of the errors caused by adding type annotations led me to adding `return None` at the end of a method. That seems like a bug (or a yak), but I'm not going to pursue it further.